File: rust_c/spatial-safety-unsafe/for-unsafe-loop.c
#include <stdio.h>

int sum(int arr[], int size) {
    int sum = 0;
    for (int i = 0; i < size; i++) {
        sum += arr[i];
    }
    return sum;
}

int main() {
    int array[] = {1, 2, 3, 4, 5, 6};
    int size = sizeof(array) / sizeof(array[0]);
    int s = sum(array, size);
    //sum += array[6];
    // Unsafe access to the element at index 6 without bounds checking
    s += array[100];
    return s;
}

File: rust_c/spatial-safety-unsafe/for-unsafe-loop.rs

 fn main(){
    let array = [1, 2, 3, 4, 5, 6];
    let mut sum = sum(&array);
    //sum += array[6];
    //sum += *array.get_unchecked(6);
    println!("Sum before is {}", sum);


    unsafe{
        // Unsafe access to the element at index 6 without bounds checking
        println!("The number is {}", *array.get_unchecked(1234));
        sum += *array.get_unchecked(1234);
    }
    println!("Sum after is {}", sum);
 }

 fn sum (arr: &[i32; 6]) -> i32{
    let mut sum = 0;
    for &num in arr {
        sum += num;
    } 
    sum
}

